dynamic_array/array.py

```diff
@@ -118,8 +118,11 @@ def __resize(self):
         """
         if self._size == self._capacity:
             new_capacity = self._capacity * 2
-        elif self._size > 0 and self._size <= self._capacity / 4:
-            new_capacity = self._capacity // 2
+        elif self._size > 0 and self._size <= self._capacity / 4 :
+            if int(self._capacity / 16) > 1:
```

Comment:
elif self._size > 0 and self._size <= self._capacity / 4 and self._capacity > 16:
        new_capacity = self._capacity // 2